src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/movements/ExternalMovementsMigrationService.kt

```diff
@@ -191,4 +200,106 @@ class ExternalMovementsMigrationService(
     nomisAddressOwnerClass = this.fromAddressOwnerClass,
     dpsAddressText = this.fromFullAddress ?: "",
   )
+
+  private fun OffenderTemporaryAbsencesResponse.toDpsRequest() = MigrateTapRequest(
```

Comment:
And this is the huge function that converts NOMIS data to the DPS migration request.